File: app/src/main/java/sumit/com/daggerdemo/ui/MainViewModel.java
package sumit.com.daggerdemo.ui;

import android.content.Context;

import sumit.com.daggerdemo.data.local.DatabaseService;
import sumit.com.daggerdemo.data.remote.NetworkService;


public class MainViewModel {

    private DatabaseService databaseService;
    private NetworkService networkService;

    public MainViewModel(Context context) {
        networkService = new NetworkService(context);
        databaseService = new DatabaseService(context);
    }

    public String getSomeData() {
        return databaseService.getDummyData() + " : " + networkService.getDummyData();
    }
}

File: app/src/main/java/sumit/com/daggerdemo/ui/MainViewModel.kt
package sumit.com.daggerdemo.ui

import android.content.Context

import sumit.com.daggerdemo.data.local.DatabaseService
import sumit.com.daggerdemo.data.remote.NetworkService


class MainViewModel(context: Context) {

    private val databaseService: DatabaseService
    private val networkService: NetworkService

    val someData: String
        get() = databaseService.dummyData + " : " + networkService.dummyData

    init {
        networkService = NetworkService(context)
        databaseService = DatabaseService(context)
    }
}
